go/libraries/doltcore/sqle/dtables/filtered_reader.go

```diff
@@ -249,18 +249,20 @@ func CreateReaderFuncLimitedByExpressions(nbf *types.NomsBinFormat, tblSch schem
 	pkCols := tblSch.GetPKCols()
 	var keySet setalgebra.Set = setalgebra.UniversalSet{}
 	var err error
-	for _, filter := range filters {
-		var setForFilter setalgebra.Set
-		setForFilter, err = getSetForKeyColumn(nbf, pkCols.GetByIndex(0), filter)
 
-		if err != nil {
-			break
-		}
+	if len(pkCols.GetColumns()) > 0 {
```

Comment:
This is fine for a bug fix but this logic ultimately needs to be ripped out and replaced with the unified indexing path that all other tables use. This was implemented back before we had a reasonable way to apply multiple filter conditions to an indexed table scan, and we do now. Just add a TODO.

Reply:
Makes sense; I added a TODO in history_table for it to stop using ThreadSafeCRFuncCache and upgrade to the new unified indexing path. ✅